### PR TITLE
feat: support VolumeExpansion in OFFLINE scenario

### LINKED ISSUE #43: support volume resize in k8s v1.14
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

CSI support is in alpha in 1.14 release and requires enabling `ExpandCSIVolumes` feature gate.
https://github.com/kubernetes/website/pull/12928
https://github.com/kubernetes/enhancements/issues/556#issuecomment-468760297

https://github.com/container-storage-interface/spec/blob/master/spec.md#nodeexpandvolume

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
support VolumeExpansion in OFFLINE scenario

**Which issue(s) this PR fixes**:

Fixes #43 

**Special notes for your reviewer**:

**Release note**:
```
feat: support VolumeExpansion in OFFLINE scenario
```

### CODE REVIEW

**Comment by ZeroMagic** [comment]
/hold
Because the current version of CSI sanity test(v1.1.0) does not support `ControllerServiceCapability_RPC_EXPAND_VOLUME`, we need to update `csi-test`.

File: pkg/azurefile/controllerserver.go [hunk 1]
```diff
@@ -321,7 +321,33 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
...
+	if capacityBytes == 0 {
+		return nil, status.Error(codes.InvalidArgument, "volume capacity range missing in request")
+	}
+	volSizeBytes := int64(capacityBytes)
+	requestGiB := int32(volumehelper.RoundUpGiB(volSizeBytes))
+
+	volumeID := req.VolumeId
+	shareURL, err := d.getShareUrl(volumeID)
+	if err != nil {
+		klog.V(4).Infof("failed to get share url with (%s): %v, returning with success", volumeID, err)
```
Comment (andyzhangx) [inline pkg/azurefile/controllerserver.go]: klog.Errorf

Comment (andyzhangx) [reply pkg/azurefile/controllerserver.go]: why returning success here? by sanity test?

File: pkg/azurefile/controllerserver.go [hunk 2]
```diff
@@ -321,7 +321,33 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
...
+		return &csi.ControllerExpandVolumeResponse{}, nil
+	}
+
+	_, err = shareURL.SetQuota(ctx, requestGiB)
```
Comment (andyzhangx) [inline pkg/azurefile/controllerserver.go]: if _, err = shareURL.SetQuota(ctx, requestGiB); err != nil {
  ...
}

File: pkg/azurefile/controllerserver.go (hunk 1 above)
Comment (ZeroMagic) [reply pkg/azurefile/controllerserver.go]: Oh, it is my fault. It should fail here.

File: pkg/azurefile/controllerserver.go (hunk 2 above)
Comment (ZeroMagic) [reply pkg/azurefile/controllerserver.go]: done.

**Comment by ZeroMagic** [comment]
I will update the version of `CSI sanity test` to v2.2.0. The number of specs increases from 57 to 72. So some new bugs appear. They will be fixed in another PR.

**Comment by andyzhangx** [comment]
btw, what's the `OFFLINE scenario`, I thought it's **online** scenario
@ZeroMagic

**Comment by ZeroMagic** [comment]
> btw, what's the `OFFLINE scenario`, I thought it's **online** scenario
> @ZeroMagic

According to the developer documentation(https://kubernetes-csi.github.io/docs/volume-expansion.html#implementing-volume-expansion-functionality)
ONLINE : Volume is currently published or available on a node.
OFFLINE : Volume is currently not published or available on a node.

So we will call ControllerExpandVolume RPC in the OFFLINE scenario, and call the NodeExpandVolume RPC in the ONLINE scenario.

**Comment by andyzhangx** [comment]
@ZeroMagic azure file should support online scenario, while azure disk only supports offline

**Pull request merged**